Runtime/AvatarCreator/Scripts/UI/Elements/BodyShapeSelectionElement.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+using System.ComponentModel;
+using System.Linq;
+using ReadyPlayerMe.Core;
+using UnityEngine;
+
+namespace ReadyPlayerMe.AvatarCreator
+{
+    /// <summary>
+    ///     This class is responsible for creating buttons to select body shapes and creating buttons for each shape.
+    ///     It extends the functionality of SelectionElement and adds selectable bodyshapes and IDs for them.
+    /// </summary>
+    public class BodyShapeSelectionElement : SelectionElement
+    {
+        [SerializeField] private AssetBodyShape[] availableBodyshapes;
+
+
+        public void LoadAndCreateButtons()
+        {
+            AddBodyShapeIds();
+            CreateButtons(availableBodyshapes.ToArray(), (button, asset) =>
+            {
+                button.SetIcon(asset.image);
+            });
+        }
+
+        public void SetAssetSelected(AvatarProperties avatarProperties)
+        {
+            if (!avatarProperties.Assets.ContainsKey(AssetType.BodyShape))
+            {
+                SetButtonSelected(GetEnumDescription(BodyShape.Average));
+                return;
+            }
+            var assetId = avatarProperties.Assets[AssetType.BodyShape] as string;
+            if (string.IsNullOrEmpty(assetId))
+            {
+                Debug.Log($"Asset id is null or empty {assetId} on type BodyShape");
+                return;
+            }
+            SetButtonSelected(assetId);
+        }
+
+        private void AddBodyShapeIds()
+        {
+            availableBodyshapes = availableBodyshapes.Select(bodyShape => new AssetBodyShape
+            {
+                bodyShape = bodyShape.bodyShape,
+                AssetType = bodyShape.AssetType,
+                image = bodyShape.image,
+                Id = GetEnumDescription(bodyShape.bodyShape)
+            }).ToArray();
+        }
+
+        private static string GetEnumDescription(Enum value)
```

Comment:
I think that it would be better to use an extension or helper method for this EG
BodyShape.GetAsID (perhaps you can think of a better name). 
```cs
public static class EnumExtensions
{
    public static string GetAsID(this Enum enumValue)
    {
        return enumValue.ToString().ToLower();
    }
}
```
The other thing is I wonder if it's not just easier (and perhaps slightly more optimized) to just get the id by something like 
```cs
BodyShape shape = BodyShape.Athletic;
string shapeName = shape.ToString().ToLower();
// shapeName = athletic
```
Instead of all this other stuff

Reply:
Hmm, not a fan of messing with IDs (upperToLower) as they should always be the same. And the description is also probably the wrong place for it. Or create a customProperty called Id and add it?

``` C#
using System;

[AttributeUsage(AttributeTargets.Field, AllowMultiple = false)]
public class IdAttribute : Attribute
{
    public string Id { get; private set; }

    public IdAttribute(string id)
    {
        Id = id;
    }
}
```

``` C#
public enum BodyShape
{
    None,
    [Id("average")] Average,
    [Id("athletic")] Athletic,
    [Id("heavyset")] HeavySet,
    [Id("plussize")] PlusSize
}
```

using System.Reflection;

```C#
public static class EnumExtensions
{
    public static string GetId(this Enum value)
    {
        var field = value.GetType().GetField(value.ToString());
        var attribute = Attribute.GetCustomAttribute(field, typeof(IdAttribute)) as IdAttribute;
        return attribute == null ? value.ToString() : attribute.Id;
    }
}
```

Reply:
Don't you think it's overcomplicating it? 
Also we use the same thing in other places 
![image](https://github.com/readyplayerme/rpm-unity-sdk-core/assets/7085672/067ed285-ead2-4106-82bf-4b190108723b)

Reply:
Example there would be the following: [Description("fullbody-xr")] FullBodyXR. This wouldn't work, since the id should be fullbody-xr. So it works for this case, but it is prone to error.

Reply:
Or we have this kind of methods
https://github.com/readyplayerme/rpm-unity-sdk-core/blob/main/Runtime/AvatarCreator/Scripts/WebRequests/AvatarAPIRequests.cs#L98-L109

Reply:
ok just leave them how they are we can review later

Reply:
Also regarding efficiency it's about twice as slow, but one call takes 0.00059 seconds, so not too worried about that.